lxd/firewall/drivers/drivers_nftables.go

```diff
@@ -405,27 +405,38 @@ func (d Nftables) InstanceSetupBridgeFilter(projectName string, instanceName str
 	}
 
 	// Filter unwanted ethernet frames when using IP filtering.
-	if IPv4 != nil || IPv6 != nil {
+	if len(IPv4Nets)+len(IPv6Nets) > 0 {
 		tplFields["filterUnwantedFrames"] = true
 	}
 
-	if IPv4 != nil {
-		if IPv4.String() == FilterIPv4All {
-			tplFields["ipv4FilterAll"] = true
-		} else {
-			tplFields["ipv4Addr"] = IPv4.String()
-		}
+	var ipv4Nets []string
+	if IPv4Nets != nil && len(IPv4Nets) == 0 {
+		tplFields["ipv4FilterAll"] = true
+	}
+	for _, ipv4Net := range IPv4Nets {
+		ipv4Nets = append(ipv4Nets, ipv4Net.String())
 	}
 
-	if IPv6 != nil {
-		if IPv6.String() == FilterIPv6All {
-			tplFields["ipv6FilterAll"] = true
-		} else {
-			tplFields["ipv6Addr"] = IPv6.String()
-			tplFields["ipv6AddrHex"] = fmt.Sprintf("0x%s", hex.EncodeToString(IPv6))
+	var ipv6Nets []map[string]string
```

Comment:
Can you move this line to just above the for loop that populates it and change it to pre-allocate the slice, e.g.

```go
ipv6Nets := make([]map[string]string, 0, len(IPv6Nets))
```